torch/distributed/distributed_c10d.py

```diff
@@ -2139,15 +2139,35 @@ def all_gather_coalesced(
     ]
 
     if group is None:
-        default_pg = _get_default_group()
-        work = default_pg.allgather_coalesced(output_tensor_lists, input_tensor_list)
-    else:
-        work = group.allgather_coalesced(output_tensor_lists, input_tensor_list)
-
+        group = _get_default_group()
+    
+    #work = group.allgather_coalesced(output_tensor_lists, input_tensor_list)
+    num_proc = len(output_tensor_lists)
+    
+    fut_list = []
+    temp_out = [[] for _ in range(len(input_tensor_list))]
+    
+    def copy_values_to_output_tensor_lists(fut_param):
+        vals = fut_param.value()
+        print(f"first:{vals} - {fut_param.done()}")
+        print(f"second: {vals[0].value()}")
+    
+    for idx, tensor in enumerate(input_tensor_list):
```

Comment:
Would I be correct if I assume this is just your tests instead of the final algorithm? Because for coalesced communication, we should just call one communication operation. (i.e., flatten inputs and outputs -> call allgather -> copy the value back to original inputs and outputs)